client.go

```diff
@@ -247,7 +248,7 @@ func (c *HTTPClient) CacheGetKeyWithStatus(keyID string, status VersionStatus) (
 	if err != nil {
 		return nil, err
 	}
-	k := Key{Path: path}
+	k := KeyAccess{Key: &Key{Path: path}}
```

Comment:
do we want to populate a value for Principal when accessing from Cache? We could use the Hostname or SPIFFE as the default?

Reply:
I believe the daemon will cache the KeyAccess object from the NetworkGetKey call: https://github.com/pinterest/knox/blob/master/client/daemon.go#L265-L289 . This means when unmarshalling it will properly unmarshal in the KeyAccess struct. This does have important considerations down the line because it breaks backwards compatibility with current Knox caches.